src/dioptra/client/_client.py

```diff
@@ -0,0 +1,707 @@
+
+        Args:
+            name: A unique string identifying a task plugin package within dioptra_builtins collection.
+
+        Returns:
+            The Dioptra REST api's response.
+
+            Example::
+
+                {
+                    'taskPluginName': 'custom_poisoning_plugins',
+                    'collection': 'dioptra_custom',
+                    'modules': ['__init__.py',
+                    'attacks_poison.py',
+                    'data_tensorflow.py',
+                    'datasetup.py',
+                    'defenses_image_preprocessing.py',
+                    'defenses_training.py',
+                    'estimators_keras_classifiers.py',
+                    'registry_art.py',
+                    'tensorflow.py']
+                }
+
+        Notes:
+           See https://pages.nist.gov/dioptra/user-guide/api-reference-restapi.html for more
+           information on Dioptra's REST api.
+        """  # noqa B950
+        task_plugin_name_query: str = urljoin(self.task_plugin_custom_endpoint, name)
+        return cast(dict[str, Any], requests.get(task_plugin_name_query).json())
+
+    def list_experiments(self) -> list[dict[str, Any]]:
+        """Gets a list of all registered experiments.
+
+        Returns:
+            A list of responses detailing all experiments.
+
+            Example::
+
+                [{
+                    'lastModified': '2023-04-24T20:20:27.315687',
+                    'experimentId': 1,
+                    'name': 'mnist',
+                    'createdOn': '2023-04-24T20:20:27.315687'
+                },
+                ...
+                {
+                    'lastModified': '2023-06-22T13:42:35.379462',
+                    'experimentId': 10,
+                    'name': 'mnist_feature_squeezing',
+                    'createdOn': '2023-06-22T13:42:35.379462'
+                }]
+
+        Notes:
+           See https://pages.nist.gov/dioptra/user-guide/api-reference-restapi.html for more
+           information on Dioptra's REST api.
+        """  # noqa B950
+        return cast(list[dict[str, Any]], requests.get(self.experiment_endpoint).json())
+
+    def list_jobs(self) -> list[dict[str, Any]]:
+        """Gets a list of all submitted jobs.
+
+        Returns:
+            A list of responses detailing all jobs.
+
+            Example::
+
+                [{
+                    'mlflowRunId': None,
+                    'lastModified': '2023-04-24T20:54:30.722304',
+                    'experimentId': 2,
+                    'queueId': 2,
+                    'workflowUri': 's3://workflow/268a7620de8247e3a00fbe466f023429/workflows.tar.gz',
+                    'entryPoint': 'train',
+                    'dependsOn': None,
+                    'status': 'queued',
+                    'timeout': '1h',
+                    'jobId': 'a4c574dd-cbd1-43c9-9afe-17d69cd1c73d',
+                    'entryPointKwargs': '-P dcd ata_dir=/nfs/data/Mnist',
```

Comment:
General question, is that an actual input for one of the examples? Usually the structure is just `"-P key=value`.